### DISCUSSION TITLE
PostgreSQL, MQTT and Grafana on separated servers

### DISCUSSION THREAD

**Discussion by edulmartins** [teslamate-org/teslamate#3564] [Q&A]
hi. I'd like to use separated server for PostgreSQL, MQTT and Grafana. Is this possible? For the MQTT looks straight forward but I'm confused is is necessary to create dockers if I won't these services 

Here is my tentative docker compose where 

version: "3"

services:
  teslamate:
    image: teslamate/teslamate:latest
    restart: always
    environment:
      - ENCRYPTION_KEY=KEY
      - DATABASE_USER=teslamate
      - DATABASE_PASS= PWD
      - DATABASE_NAME=teslamate
      - DATABASE_HOST=192.168.12.207
      - MQTT_HOST=192.168.12.202
    ports:
      - 4000:4000
    volumes:
      - ./import:/opt/app/import
    cap_drop:
      - all

  database:
    image: postgres:16
    restart: always
    environment:
      - POSTGRES_USER=teslamate
      - POSTGRES_PASSWORD=PWD
      - POSTGRES_DB=teslamate
    volumes:
      - teslamate-db:/var/lib/postgresql/data

  grafana:
    image: teslamate/grafana:latest
    restart: always
    environment:
      - DATABASE_USER=teslamate
      - DATABASE_PASS=PWD
      - DATABASE_NAME=teslamate
      - DATABASE_HOST=database
    ports:
      - 3000:3000
    volumes:
      - teslamate-grafana-data:/var/lib/grafana

  mosquitto:
    image: eclipse-mosquitto:2
    restart: always
     environment:
      - MQTT_USERNAME=teslamate
      - MQTT_PASSWORD=PWD
    ports:
      - 1883:1883
    volumes:
      - mosquitto-conf:/mosquitto/config
      - mosquitto-data:/mosquitto/data

volumes:
  teslamate-db:
  teslamate-grafana-data:
  mosquitto-conf:
  mosquitto-data:

**Comment by litithio** [comment]
I got the same issue. I already run a mosquitto and a grafana container for other use cases. I don't want to run these services twice. Any possibility to use existing containers would be great.
